rholang/examples/vault_demo/demo_script.md

```diff
@@ -2,68 +2,88 @@
 The only things needed are access to your RevVault, and the target RevAddress.
 
-    ./propose.sh $ALICE_PRV vault_demo/3.transfer_funds.rho "-e s/%REV_ADDR/$BOB_REV/"
-    ./propose.sh $ALICE_PRV vault_demo/2.access_your_own_vault.rho
+*In this example we're deploying as a Testnet operator, but it works exactly the same for a normal user.*
+
+    ./propose.sh $GENESIS_PRV vault_demo/3.transfer_funds.rho "-e s/%FROM/$GENESIS_REV/ -e s/%TO/$ALICE_REV/"
+    ./propose.sh $ALICE_PRV vault_demo/2.check_balance.rho "-e s/%REV_ADDR/$ALICE_REV/"
+
+Notice that anybody can query the balance of any RevVault.
+
+    ./propose.sh $GENESIS_PRV vault_demo/2.check_balance.rho "-e s/%REV_ADDR/$ALICE_REV/"
 
-Notice that an exchange would do the transfer exactly the same way.
+An exchange would do the transfer exactly the same way.
+That's also exactly what the Testnet operators are going to do to distribute initial funds.
 
 Because the "transfer" method takes a RevAddress (and not a RevVault),
 transfers between different "kinds", or security schemes of RevVaults are possible.
 
-For now, we only provide a simple Lockbox that only grants access to its designated user.
-By "user" here we mean the deployer of the contract that uses the Lockbox and its underlying RevVault.
-    
-Notice that the transfer was initiated, but hasn't finished yet. Have burnt the REV?     
+For now, we only provide a simple RevVault that only grants access to its designated user.
+By "user" here we mean the deployer of the contract that uses the RevVault and its corresponding AuthKey.
 
 ## Onboard a friend on the platform
 
-*This is subject to change, it might be that the platform will perform this step automatically during users first deploy.* 
+    ./propose.sh $ALICE_PRV vault_demo/3.transfer_funds.rho "-e s/%FROM/$ALICE_REV/ -e s/%TO/$BOB_REV/"
+    ./propose.sh $ALICE_PRV vault_demo/2.check_balance.rho "-e s/%REV_ADDR/$ALICE_REV/"
+
+Notice the transfer hasn't been finished yet. Still, funds have been deducted from Alice's vault.
+
+Let's execute as Bob and check his balance:
 
-    ./propose.sh $ALICE_PRV vault_demo/1.create_vault.rho "-e s/%PUB_KEY/$BOB_PUB/"
+    ./propose.sh $BOB_PRV vault_demo/2.check_balance.rho "-e s/%REV_ADDR/$BOB_REV/"
 
 Notice the transfer that was initiated previously has now been finished.
-Let's become Bob and check the balance. 
 
-    ./propose.sh $BOB_PRV vault_demo/2.access_your_own_vault.rho
+## There's been a race
+
+You might've been lucky to observe a **race** between the balance check and the transfer arrival.
+So it might be that you've seen "Balance is 0" followed by "Finished transfer" instead of
+"Finished transfer" followed by "Balance is 100". This is OK though, the next deploy will no longer
+be subject to this race, as Bob's vault exists after the balance check has been done.
+
+This also means that the first access to one's vault needs to be done by a 3rd-party having the REV
```

Comment:
In practice, nobody is going to pay REV to check a balance, thus creating wallets. It's a readonly operation and we will build tools to do this on readonly nodes for free.